### ISSUE TITLE
Environment variables from `.env` are not picked up by the Node process

### ISSUE THREAD

**Issue by beeman** [nrwl/nx#12108]
## Current Behavior
With the latest version of nx, it seems like the env vars in `.env` are no longer picked up in Node/Nest apps (possibly others).

Some of the env vars do appear to be set:

```ts
console.log(Object.keys(process.env))
[
  'NODE_ENV',
  'NX_CLI_SET',
  'NX_INVOKED_BY_RUNNER',
  'NX_WORKSPACE_ROOT',
  'NX_TERMINAL_OUTPUT_PATH',
  'NX_STREAM_OUTPUT',
  'NX_TASK_TARGET_PROJECT',
  'NX_TASK_HASH'
]
```

## Expected Behavior
Env vars defined in `.env` should be available on `process.env` 

## Steps to Reproduce

Repro here: https://github.com/beeman/nx-env-vars-issue

- create a new repo
- add  @nrwl/nest or  @nrwl/node and create an app
- add an env var to `.env` 
- log out `process.env`
- see that the env var from `.env` is not there

### Failure Logs
No logs

### Environment

```
 >  NX   Report complete - copy this into the issue template

   Node : 16.16.0
   OS   : darwin arm64
   yarn : 1.22.19

   nx : 14.7.6
   @nrwl/angular : Not Found
   @nrwl/cypress : Not Found
   @nrwl/detox : Not Found
   @nrwl/devkit : 14.7.6
   @nrwl/eslint-plugin-nx : 14.7.6
   @nrwl/expo : Not Found
   @nrwl/express : Not Found
   @nrwl/jest : 14.7.6
   @nrwl/js : 14.7.6
   @nrwl/linter : 14.7.6
   @nrwl/nest : 14.7.6
   @nrwl/next : Not Found
   @nrwl/node : 14.7.6
   @nrwl/nx-cloud : 14.6.2
   @nrwl/nx-plugin : Not Found
   @nrwl/react : Not Found
   @nrwl/react-native : Not Found
   @nrwl/schematics : Not Found
   @nrwl/storybook : Not Found
   @nrwl/web : Not Found
   @nrwl/workspace : 14.7.6
   typescript : 4.8.3
   ---------------------------------------
   Local workspace plugins:
   ---------------------------------------
   Community plugins:
```

**Comment by yamadayuki** [comment]
I face the same problem. `@nrwl/webpack:webpack` executor uses the webpack config derived from [`getBaseWebpackPartial`](https://github.com/nrwl/nx/blob/f49769a34a7f47d252648132793cddd2612262ee/packages/webpack/src/utils/config.ts#L34-L35). This config has the `webpack.DefinePlugin` instance with `getClientEnvironment`.

https://github.com/nrwl/nx/blob/f49769a34a7f47d252648132793cddd2612262ee/packages/webpack/src/utils/config.ts#L131-L132

**Comment by jimbuck** [comment]
I am seeing this specifically with environment variables **not** prefixed with `NX_` (I'm using NX 14.7.6 with Express).

**Comment by beeman** [comment]
> I am seeing this specifically with environment variables **not** prefixed with `NX_` (I'm using NX 14.7.6 with Express).

You are right, those do appear. I'll update my post, thanks!

**Comment by ratoi-crysty** [comment]
I can confirm that this issue is also present with Nest as well regarding environment variables (provided from CLI).

**Comment by Michsior14** [comment]
Quick workaround without downgrade if you use custom webpack config:

```ts
module.exports = (config, _) => {
    // your code ...
    config.plugins = config.plugins.filter((p) => p.constructor.name !== 'DefinePlugin');
    return config;
}
```

**Comment by mahnuh** [comment]
This is still happening for me with 14.7.8, should it have been fixed?

**Comment by Michsior14** [comment]
The fix landed on master, but it's not released yet. I guess it will be available in 14.7.9 ;)
